signer/src/eth.rs

```diff
@@ -0,0 +1,346 @@
+// Copyright 2019-2024 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+//! An ethereum keypair implementation.
+
+use derive_more::{Display, From};
+use hex::FromHex;
+use keccak_hash::keccak;
+use secp256k1::{Message, Secp256k1, SecretKey};
+
+use crate::crypto::{DeriveJunction, SecretUri};
+use crate::ecdsa;
+
+const SEED_LENGTH: usize = 32;
+
+/// Seed bytes used to generate a key pair.
+pub type Seed = [u8; SEED_LENGTH];
+
+/// An ethereum keypair implementation.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Keypair(ecdsa::Keypair);
+
+impl From<ecdsa::Keypair> for Keypair {
+    fn from(kp: ecdsa::Keypair) -> Self {
+        Self(kp)
+    }
+}
+
+impl Keypair {
+    /// Construct a keypair from a hex-encoded private key.
+    pub fn from_private_key_hex(hex: &str) -> Result<Self, Error> {
```

Comment:
Technically this API supports `"<64 hex digits>"` but not `"0x<64 hex digits>`.

I think we should remove this because it's better to have a uniform API because [SecretUri](https://docs.rs/subxt-signer/latest/subxt_signer/struct.SecretUri.html) already supports hex strings with the format "0x<64 hex digits>".

Reply:
Yeah, let's remove this; the right way should be to do any hex stuff yourself and then call `from_seed` on the 32 bytes.